src/azul/indexer/__init__.py

```diff
@@ -174,9 +174,21 @@ class SourceSpec(ABC, Generic[SOURCE_SPEC]):
     prefix: Prefix
 
     @classmethod
-    @abstractmethod
     def parse(cls, spec: str) -> SOURCE_SPEC:
-        raise NotImplementedError
+        _spec, sep, prefix = spec.rpartition(':')
+        reject(sep == '',
+               'Source specifications must end in a colon followed by an optional UUID prefix')
+        prefix = Prefix.parse(prefix)
+        if _spec.startswith('tdr'):
+            # Import locally to avoid cyclical import
+            from azul.terra import (
+                TDRSourceSpec,
+            )
+            self = TDRSourceSpec._parse(_spec, prefix)
```

Comment:
@hannes-ucsc I think this is close to what we discussed... The parent class being aware of the subclass is a bit strange, but i guess we can live with it. As an alternative we could:

return a SourceSpec (which is an ABC) to the subclasses .parse method, then attr.evolve the passed instance into a instance of the appropriate subclass.... 
Not 100% sure if this would work just a thought.